Refactor: Extract effect handling logic into separate functions and Kotlin member ordering

Mostafa-alsaygh opened the pull request:
This PR refactors several screens to extract the logic for handling UI effects into separate private composable functions. This improves code organization and readability by separating concerns.

The following screens were affected:
- `ActorDetailsScreen`
- `ContinueWatchingScreen`
- `HomeScreen`
- `LoginScreen`
- `TopRatedScreen`

Additionally, the following changes were made:
- Follow Kotlin coding conventions for member ordering
- `TvShowDetailsContract.kt`: Added `onEpisodeClick` function to the interface.
- `EpisodeDetailsViewModel.kt`: Reordered `onBackClicked` function.
- `NovixApp.kt`: Modified `onNavigateBack` in `LoginScreen` to navigate to `Screen.OnBoarding.Welcome`.
- `SearchScreen.kt`: Minor code formatting.
- Removed unnecessary comments from TvShowDetailsScreen.
- Adjusted top padding for the back button in ActorDetailsScreen.

Mostafa-alsaygh committed:
Refactor: Extract effect handling logic into separate functions

This commit refactors several screens to extract the logic for handling UI effects into separate private composable functions. This improves code organization and readability by separating concerns.

The following screens were affected:
- `ActorDetailsScreen`
- `ContinueWatchingScreen`
- `HomeScreen`
- `LoginScreen`
- `TopRatedScreen`

Additionally, the following changes were made:
- Follow Kotlin coding conventions for member ordering
- `TvShowDetailsContract.kt`: Added `onEpisodeClick` function to the interface.
- `EpisodeDetailsViewModel.kt`: Reordered `onBackClicked` function.
- `NovixApp.kt`: Modified `onNavigateBack` in `LoginScreen` to navigate to `Screen.OnBoarding.Welcome`.
- `SearchScreen.kt`: Minor code formatting.

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/feature/continuewatching/ContinueWatchingScreen.kt:
```diff
@@ -160,6 +159,23 @@ fun Content(
 }
 
+@Composable
+private fun HandleContinueWatchingEffects(
```
you are the only one in the squad that follows this approach, kindly use our approach, Effect.listen

Mostafa-alsaygh committed:
Refactor: Reorder methods in SearchViewModel and ReviewsViewModel

AbdelrhmanAyman20470 commented on presentation/src/main/java/com/london/presentation/feature/continuewatching/ContinueWatchingScreen.kt:
```diff
@@ -160,6 +159,23 @@ fun Content(
+    effect: ContinueWatchingEffect?,
+    onMovieClick: (Int) -> Unit,
+    onTvShowClick: (Int) -> Unit,
+    onBackClick: () -> Unit
```
rename it to onBack because its clear that button for click back

AbdelrhmanAyman20470 commented on presentation/src/main/java/com/london/presentation/feature/details/actor/ActorDetailsScreen.kt:
```diff
@@ -281,6 +274,33 @@ fun ActorScreenContent(
+@Composable
+private fun HandleActorDetailsScreenEffects(
+    effect: ActorEffectUiState?,
+    uiState: ActorDetailsUiState, // Assuming this is your state type
```
not need this comment

AbdelrhmanAyman20470 commented on presentation/src/main/java/com/london/presentation/feature/details/actor/ActorDetailsScreen.kt:
```diff
@@ -281,6 +274,33 @@ fun ActorScreenContent(
+private fun HandleActorDetailsScreenEffects(
+    effect: ActorEffectUiState?,
+    uiState: ActorDetailsUiState, // Assuming this is your state type
+    onNavigateBack: () -> Unit,
```
rename it to onBack because its clear that button for click back

Mostafa-alsaygh committed:
Refactor: Standardize back navigation parameter name

The `onNavigateBack` parameter in `ActorDetailsScreen` and related navigation functions has been renamed to `onBack` for consistency with other composables.

AbdoMagdey24 commented on app/src/main/java/com/london/app/navigation/NovixApp.kt:
```diff
@@ -201,7 +201,7 @@ fun NavGraphBuilder.authNavGraph(
             },
             onNavigateBack = {
-                navController.popBackStack()
+                navController.navigate(Screen.OnBoarding.Welcome)
```
Why ?

Mostafa-alsaygh replied:
the click of onBack did not work before, for disable the pop back because  prevent the home from back to welcome screen or onboarding

london-bot[bot] committed:
Branch was auto-updated.

marwanqashwa commented on presentation/src/main/java/com/london/presentation/feature/details/tvshow/tvshowdetails/TvShowDetailsViewModel.kt:
```diff
@@ -41,6 +41,32 @@ class TvShowDetailsViewModel(
 
     }
 
+    override fun onEpisodeDetailsClicked(tvShowId: Int, episodeNumber: Int, seasonNumber: Int) {
```
Why you delete and add methods or move her place in this file TV show details view model

Mostafa-alsaygh replied:
i said in the discerption 

- Follow Kotlin coding conventions for member ordering

- TvShowDetailsContract.kt: Added onEpisodeClick function to the interface.

marwanqashwa commented on presentation/src/main/java/com/london/presentation/feature/home/HomeScreen.kt:
```diff
@@ -313,11 +338,13 @@ private fun LazyGridScope.upComingSection(
+            Box(
+                modifier = Modifier
+                    .height(20.dp)
+                    .padding(bottom = 4.dp)
```
Please take care from your formater many files show you delete and add  because change the format

Mostafa-alsaygh replied:
the code is not clean

london-bot[bot] committed:
Branch was auto-updated.

Mostafa-alsaygh committed:
Merge branch 'develop' into refactor/presentation

# Conflicts:
#	presentation/src/main/java/com/london/presentation/feature/details/tvshow/tvshowdetails/TvShowDetailsViewModel.kt

Mostafa-alsaygh committed:
Refactor: Clean up TvShowDetailsScreen and TvShowDetailsViewModel

yusufnasserdev commented on presentation/src/main/java/com/london/presentation/feature/home/HomeScreen.kt:
```diff
@@ -77,22 +77,22 @@ fun HomeScreen(
+    )
 
     val lifecycleOwner = LocalLifecycleOwner.current
     LaunchedEffect(key1 = Unit) {
```
key1 is redundant.

london-bot[bot] committed:
Branch was auto-updated.

Mostafa-alsaygh committed:
Refactor: Standardize navigation parameter name to `onNavigateBack`

Mostafa-alsaygh committed:
Merge remote-tracking branch 'origin/refactor/presentation' into refactor/presentation

Mostafa-alsaygh committed:
Refactor: Standardize navigation parameter name to `onNavigateBack`
